tests/run-e2e.sh

```diff
@@ -18,8 +25,155 @@ declare SHOW_USAGE=false
 declare LOGS_DIR="tmp/e2e"
 declare OPERATORS_NS="operators"
 declare TEST_TIMEOUT="15m"
-declare BUNDLE_IMG="quay.io/sustainable_computing_io/kepler-operator-bundle"
-declare OPERATOR_IMG="quay.io/sustainable_computing_io/kepler-operator"
+# declare -a PORT_FORWARDED_PIDS=()
+
+cleanup() {
+	info "Cleaning up ..."
+	# shell check  ignore word splitting when using jobs -p
+	# shellcheck disable=SC2046
+	[[ -z "$(jobs -p)" ]] || kill $(jobs -p) || true
+
+	return 0
+}
+
+delete_olm_subscription() {
+	header "Delete Old Deployments"
+
+	$CI_MODE && {
+		ok "skipping deletion of old deployment in CI mode"
+		return 0
+	}
+
+	kubectl delete -n "$OPERATORS_NS" csv --all || true
+	kubectl delete -n "$OPERATORS_NS" installplan,subscriptions,catalogsource \
+		-l operators.coreos.com/$OPERATOR.operators= || true
+	kubectl delete -n "$OPERATORS_NS" installplan,subscriptions,catalogsource \
+		-l operators.coreos.com/$OPERATOR.openshift-operators= || true
+	kubectl delete -n "$OPERATORS_NS" catalogsource $OLM_CATALOG || true
+}
+
+build_bundle() {
+	header "Build Operator Bundle"
+
+	$NO_BUILDS && {
+		info "skipping building of images"
+		return 0
+	}
+
+	run make operator-build bundle bundle-build \
+		OPERATOR_IMG="$OPERATOR_IMG" \
+		BUNDLE_IMG="$BUNDLE_IMG" \
+		VERSION="$VERSION"
+}
+
+push_bundle() {
+	header "Push Operator Bundle Images"
+	$NO_BUILDS && {
+		info "skipping pushing images"
+		return 0
+	}
+
+	run make operator-push bundle-push \
+		OPERATOR_IMG="$OPERATOR_IMG" \
+		BUNDLE_IMG="$BUNDLE_IMG" \
+		VERSION="$VERSION"
+
+}
+
+run_bundle() {
+	header "Running Bundle"
+
+	run ./tmp/bin/operator-sdk run bundle "$BUNDLE_IMG" \
+		--install-mode AllNamespaces --namespace "$OPERATORS_NS" --skip-tls
+}
+
+log_events() {
+	local ns="$1"
+	shift
+	kubectl get events -w \
+		-o custom-columns=FirstSeen:.firstTimestamp,LastSeen:.lastTimestamp,Count:.count,From:.source.component,Type:.type,Reason:.reason,Message:.message \
+		-n "$ns" | tee "$LOGS_DIR/$ns-events.log"
+}
+
+watch_operator_errors() {
+	local err_log="$1"
+	shift
+
+	kubectl logs -f -n "$OPERATORS_NS" "deploy/$OPERATOR_DEPLOY_NAME" |
```

Comment:
We shouldn't use -f here.

Reply:
we should use -f here :) .. it looks only for errors. See the invocation below where this is run in background.